spinner added to wall and statistics pages

ulaulala opened the pull request:
According to #7 

Spinner svg is my own project so it's totally free for sure :) I will improve it later, but I'm not a designer so please #nohate 

Component Spinner is based on styled-components documentation.
[source: https://www.styled-components.com/docs/basics#animations]

Spinnners are added to Wall and Statistics pages with objects destructing.

Tuhaj commented on src/assets/spinner.svg:
```diff
@@ -0,0 +1,7 @@
+<circle cx="96" cy="43" r="12" fill="white"/>
+<circle cx="95" cy="73" r="11" fill="white"/>
+<circle cx="79" cy="94" r="9" fill="white"/>
+</svg>
```
no indentation!

dotadota replied:
Consider using svgo to optimize svg file :)

https://github.com/svg/svgo

kgajowy replied:
you may want to consider grouping the filling color:
```
<svg width="110" height="110" viewBox="0 0 110 110" fill="none" xmlns="http://www.w3.org/2000/svg">
  <g fill="white">
    <circle cx="57.5" cy="102.5" r="7.5"/>
    <circle cx="37" cy="101" r="6"/>
    <circle cx="96" cy="43" r="12"/>
    <circle cx="95" cy="73" r="11"/>
    <circle cx="79" cy="94" r="9"/>
  </g>
</svg>
```
Furthermore, assuming you will be `styled-componenting` svg, you can further make the SVG being more React-like, for example with fill color being provided via css
```
const Svg = styled.svg`
 color: rgba(255, 255, 255, 1);
`
// then, in the svg alone:
fill="currentColor"
```

ulaulala replied:
@dotadota should I optimize this 7 simple lines of svg? I don't see any prop which is unnecessary in this file. Please tell me why I need it now

Tuhaj commented on src/components/Spinner.js:
```diff
@@ -0,0 +1,36 @@
+   }
+`
+
+export default Spinner;
```
new line

ulaulala replied:
Yes, it's a new line.

kgajowy replied:
We shall consider using husky & lint-staged & prettier to keep our Code Review meaningful. 
https://github.com/kgajowy/community-suggestions-web/blob/master/.lintstagedrc
https://github.com/kgajowy/community-suggestions-web/blob/master/.huskyrc
This is a great topic for different PR which would save us a few moments when talking/writing about formatting issues ;)

kgajowy commented on src/components/ListOfCards.js:
```diff
@@ -2,24 +2,29 @@ import React, { Component } from 'react';
                 {
-                    this.state.kudoses.map((kudos, index) => <Card kudos={kudos} key={index}/>)
+                    loading ? <Spinner/> :
+                    kudoses.map((kudos, index) => <Card kudos={kudos} key={index}/>)
```
it is a good practice to use real ID instead of index. Thanks to that, React will know if the given element is the same element or a new one.
```
// array like
<Card key={0}/>
<Card key={1}/>

// ID like
<Card key={'34eqr4tzcv-random-id-which-may-be-a-string-or-number'}/>
```

https://reactjs.org/docs/lists-and-keys.html#keys

```We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.```

ulaulala replied:
What if I make key like a combination of kudos 'name', 'description' and 'from' property (it's all data which I have from backend), and someone add one kudos more than once? Keys will be duplicated. Maybe we could add an ID property to object to make keys combination unique?

kgajowy replied:
@ulaulala backend shall return you the ID; as you noticed making self-constructed ID will not grant the uniquity

ulaulala replied:
Ok, I will make an issue for that

kgajowy commented on src/components/ListOfCards.js:
```diff
@@ -2,24 +2,29 @@ import React, { Component } from 'react';
+        kudoses: [],
+        loading: true
     }
     componentWillMount () {
```
This kind of activity should take place in `componentDidMount`. The reason is that the `then` may happen (not likely to happen very often with the real API) before component is mounted, thus trying to `setState` and further change the component which is not yet mounted.
Due to this, React Team decided to drop this lifecycle method:
https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

```
UNSAFE_componentWillMount() is invoked just before mounting occurs. It is called before render(), therefore calling setState() synchronously in this method will not trigger an extra rendering. Generally, we recommend using the constructor() instead for initializing state.

Avoid introducing any side-effects or subscriptions in this method. For those use cases, use componentDidMount() instead.
```

ulaulala commented:
Guys, code improvements added. I think everything's good now. I decided to not add svgo, because is unnecessary now. I moved svg code to component to make faster visible and changed componentWillMount to componentDidMount. Keys will be improved later - when backend will return IDs of kudoses. Empty lines are good for me because they are natural separators.
